src/main/asciidoc/transactions.adoc

```diff
@@ -1,114 +1,191 @@
-    @Override public String getUserName() { /* ... */ }
-    @Override public String getPassword() { /* ... */ }
-    @Override public String getBucketName() { /* ... */ }
-
-	@Bean
-	public Transactions transactions(final Cluster couchbaseCluster) {
-		return Transactions.create(couchbaseCluster, TransactionConfigBuilder.create()
-			// The configuration can be altered here, but in most cases the defaults are fine.
-			.build());
-	}
-
+// Usual Setup
+@Override public String getConnectionString() { /* ... */ }
+@Override public String getUserName() { /* ... */ }
+@Override public String getPassword() { /* ... */ }
+@Override public String getBucketName() { /* ... */ }
+
+// Customization of transaction behavior is via the configureEnvironment() method
+@Override
+public void configureEnvironment(ClusterEnvironment.Builder builder) {
+  // allow really long transactions
+  builder.transactionsConfig(
+    TransactionsConfig.builder().timeout(Duration.ofSeconds(30)));
+  }
 }
 ----
 ====
 
-Once the `@Bean` is configured, you can autowire it from your service (or any other class) to make use of it. Please
-see the https://docs.couchbase.com/java-sdk/3.0/howtos/distributed-acid-transactions-from-the-sdk.html[Reference Documentation]
-on how to use the `Transactions` class. Since you need access to the current `Collection` as well, we recommend you to also
-autowire the `CouchbaseClientFactory` and access it from there:
+
+Functioning of the  @Transactional method annotation requires 
+[start=1]
+. the configuration class to be annotated with @EnableTransactionManagement;
+. the service object with the annotated methods must be annotated with @Service and @Component;
+. the service object with the annotated methods must be obtained as an @Bean either via @Autowire or
+from the application context;
+. the call to the method must be made from a different class than service because calling an annotated
+method from the same class will not invoke the Method Interceptor that does the transaction processing.
+
+== Transactions with @Transactional
+
+@Transactional Defines as transactional a method or all methods on a class.
+ 
+When this annotation is declared at the class level, it applies as a default
+to all methods of the declaring class and its subclasses. Note that it does not
+apply to ancestor classes up the class hierarchy; inherited methods need to be
+locally redeclared in order to participate in a subclass-level annotation. For
+details on method visibility constraints, consult the
+https://docs.spring.io/spring-framework/docs/current/reference/html/data-access.html#transaction[Transaction Management]
+section of the reference manual.
+ 
+===  Attribute Semantics
+ 
+In this release, he Couchbase Transactions implementation in 
+Spring Data Couchbase ignores all @Transactional attributes.
+The transaction will roll back on uncaught exceptions. 
+The transaction isoloation level is read committed;
```

Comment:
Lot of typos here.

And "In this release, he Couchbase Transactions implementation in Spring Data Couchbase ignores all @Transactional attributes." - it doesn't.  I made it support most of them - timeout, propagation levels, etc.  I think it's only rollback rules we don't.

Reply:
Well we reject at runtime non-supported isolation levels, but I think that's allowed.

Reply:
Heh :)